### PR TITLE
docs: add Web Developer Tooling use case

### PR DESCRIPTION
**Pull request by bmeurer**
This was identified as missing in the last WG meeting, and it's planned
to be addressed by the upcoming WebToolingBenchmark later this year
(see #138).

### CODE REVIEW

File: docs/use_cases.md [hunk 1]
```diff
@@ -75,6 +76,15 @@ the system or the applications running in it as directed by
 an external monitoring/management system.  The ability to be updated
 remotely is often a key requirement.
 
+## Web Developer Tooling
+
+Node.js is extensively used in all kinds of tools commonly used
+to build modern web pages, including but not limited to projects like
+[TypeScript](https://www.typescriptlang.org/), [webpack](http://webpack.js.org),
+[Babel](http://babeljs.io), and [ESLint](http://eslint.org). All
```
Comment (kunalspathak) [inline docs/use_cases.md]: I would also include 1 testing framework `mocha` or `chai`.

Comment (bmeurer) [reply docs/use_cases.md]: Done, thanks.

**Comment by bmeurer** [comment]
Anything else missing for this PR?

**Comment by mhdawson** [comment]
Landed as 21ca819d15621f94efe88b8eaad3cecadb160f36. Thanks for taking the time to add that use case.

**Pull request closed without merging**